extension/intents/tabs/tabs.js

```diff
@@ -597,3 +598,166 @@ intentRunner.registerIntent({
     context.displayText(message);
   },
 });
+
+intentRunner.registerIntent({
+  name: "tabs.selectAllTabs",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const tabs = await browser.tabs.query({ currentWindow: true});
+    const allWindows = await browser.windows.getAll({
+      windowTypes: ["normal"],
+    });
+
+    const matchingTabs = await getMatchingTabs({
+      tabs,
+      activeTab,
+      sort_by_index: true,
+      allWindows: context.parameters.allWindows === "false"
+    });
+
+    if (matchingTabs.length === 0) {
+      const exc = new Error("No tab that matches the query");
+      exc.displayMessage = "There is no tab that matches the query";
+      throw exc;
+    }
+
+    await selectAllTabs(matchingTabs);
+
+  },
+});
+
+async function selectAllTabs(tabs){
+  if(tabs.length !== 0){
+    browser.tabs.query({
+      currentWindow: true
+    }).then((tabs) => {
```

Comment:
This will be easier to read (and safer in case of exceptions) if you use `await`, like:

```js
tabs = await browser.tabs.query({currentWindow: true});
```

Also this should also use `pinned: false`, as most things that handle "all" tabs only handle the not-pinned tabs.